perl-package/AI-MXNetCAPI/mxnet_typemaps.i

```diff
@@ -112,6 +112,32 @@
     }
 }
 
+%typemap(in,numinputs=0) (const int **out_stypes) (int* temp)
+{
+    temp = NULL;
+    $1 = &temp;
+}
+
+%typemap(argout) (const int **out_stypes)
+{
+    if(av_len((AV*)SvRV(ST(3))) == -1 && !result)
+    {
+        AV *myav;
+        SV **svs;
+        int i = 0;
+        svs = (SV **)safemalloc(*arg4*sizeof(SV *));
+        for (i = 0; i < *arg4 ; i++) {
+            svs[i] = newSViv((*$1)[i]);
+            sv_2mortal(svs[i]);
+        }
+        myav = av_make(*arg4, svs);
```

Comment:
I went with this solution because I needed to solve a peculiar problem of making a decision dependent on a particular input state.
You helped me in the past to clean up the swig usage, could you please see if you can come up with a solution that is more stable (I used this technique in another place as well in the past).

Reply:
okay, sure.  I'll get more details and work out a proposal.